src/dials/algorithms/correlation/analysis.py

```diff
@@ -87,9 +93,16 @@ def __init__(
         )
 
         # Used for optional json creation that is in a format friendly for import and analysis (for future development)
-        self.ids_to_identifiers_map = {}
-        for table in self._reflections:
-            self.ids_to_identifiers_map.update(table.experiment_identifiers())
+        # Also to retain dataset ids when used in multiplex
+        if self.ids_to_identifiers_map is None:
+            self.ids_to_identifiers_map = {}
+            for table in self._reflections:
+                self.ids_to_identifiers_map.update(table.experiment_identifiers())
+
+        self.labels = list(dict.fromkeys(self.ids_to_identifiers_map))
+        self._labels_all = flex.size_t()
+        for i in self.labels:
+            self._labels_all.extend(flex.size_t(1, i))
```

Comment:
I think there is a slightly more efficient way to do this than n extends, perhaps `self._labels_all = flex.size_t(self.labels)` ?

Reply:
Have changed to the suggested code - thanks for the suggestion :)